internal/authentication/password_hash.go

```diff
@@ -123,7 +123,11 @@ func HashPassword(password, salt string, algorithm CryptAlgo, iterations, memory
 	}
 
 	if salt == "" {
-		salt = utils.RandomString(saltLength, HashingPossibleSaltCharacters)
+		if algorithm == HashingAlgorithmArgon2id {
+			salt = crypt.Base64Encoding.EncodeToString([]byte(utils.RandomString(saltLength, HashingPossibleSaltCharacters)))
+		} else {
+			salt = utils.RandomString(saltLength, HashingPossibleSaltCharacters)
```

Comment:
Since we've found that sha512 must be b64 valid we can probably simplify this code with 
```
if salt == "" {
      		salt = crypt.Base64Encoding.EncodeToString([]byte(utils.RandomString(saltLength, HashingPossibleSaltCharacters)))
	}
```